### PR TITLE
Add vector math tutorial

### PR DESCRIPTION
**Pull request by Perryvw**

### CODE REVIEW

**Comment by Shushishtok** [comment]
Thanks Perry for the great guide! Very well done.

Here are a few comments for those. I'm looking at it from a perspective of someone completely new to Vectors, so sometimes something is obvious to those familiar but isn't that very clear to newcomers.

In the "What is a Vector" section, the note says:
> Note: vectors have no origin and always originate from (0, 0).
You might want to add an introduction to what (0,0), and why vectors always originate from it.

In the "Subtracting vectors" section, add an example to a "real time" example of using vector subtractions this way.

In the "Vectors as Direction/Orientation" section, add a brief introduction to cos and sin; not everyone knows what they represent.

In the "Dot product" section, the note is not clear; it says to watch out for non-length 1 vectors, but why exactly? Is there any reason we would dot-product any non-length 1 vector in the first place?

In the "Spawning an item in front of the player" section, add an explanation to the code. Why does that line of code works? Maybe add a few graphs showing how that works in a "step by step" for that calculation.

In the "Creating some effects around player" section, add a definition for pi by running:
`const pi = Math.PI`
Also, show examples for Typescript, Javascript and lua, which are likely languages to use this for moddota comers.

Great guide overall, very well done!

File: _articles/scripting/vector-math.md [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+const angle = 2 * Math.PI / numPoints;
+
+for (let i = 0; i < numPoints; i++) {
+    // Create direction vector from the angle
+    const direction = Vector(math.cos(angle * i), math.sin(angle * i));
+    // Multiply the direction (length 1) with the desired radius of the circle
+    const circlePoint = direction * radius as Vector;
+
+    // Add the calculated green vector to the player position and do something
+    doSomething(player:GetAbsOrigin() + circlePoint);
```
Comment (ark120202) [inline _articles/scripting/vector-math.md:265]:
```suggestion
    doSomething(player.GetAbsOrigin() + circlePoint);
```

File: _articles/scripting/vector-math.md [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+```
+
+```ts
+// Calculate the angle between each point on the circle
+// (This is in radians, the full circle is 2*pi radians)
+const angle = 2 * Math.PI / numPoints;
+
+for (let i = 0; i < numPoints; i++) {
+    // Create direction vector from the angle
+    const direction = Vector(math.cos(angle * i), math.sin(angle * i));
```
Comment (ark120202) [inline _articles/scripting/vector-math.md:260]:
```suggestion
    const direction = Vector(Math.cos(angle * i), Math.sin(angle * i));
```

File: _articles/scripting/vector-math.md [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+    -- Next we update the projectile position by simply adding the velocity
+    projectile.position = projectile.position + projectile.velocity
+end
+```
+
+```ts
+function updateProjectile(projectile: Projectile, target: CDOTA_BaseNPC): void {
+  // Calculate direction from projectile to target
+  let relativeTargetPos = (target.GetAbsOrigin() - projectile.GetAbsOrigin()) as Vector;
+  let targetDirection = relativeTargetPos.Normalized();
```
Comment (ark120202) [inline _articles/scripting/vector-math.md:311]:
```suggestion
  const relativeTargetPos = (target.GetAbsOrigin() - projectile.GetAbsOrigin()) as Vector;
  const targetDirection = relativeTargetPos.Normalized();
```

File: _articles/scripting/vector-math.md [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+const angle = (2 * Math.PI) / numPoints;
+
+for (let i = 0; i < numPoints; i++) {
+  // Create direction vector from the angle
+  const direction = Vector(Math.cos(angle * i), Math.sin(angle * i));
+  // Multiply the direction (length 1) with the desired radius of the circle
+  const circlePoint = (direction * radius) as Vector;
+
+  // Add the calculated green vector to the player position and do something
+  doSomething(player.GetAbsOrigin() + circlePoint);
```
Comment (ark120202) [inline _articles/scripting/vector-math.md:273]:
```suggestion
  doSomething((player.GetAbsOrigin() + circlePoint) as Vector);
```

File: _articles/scripting/vector-math.md [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+
+<MultiCodeBlock group="vscripts">
+
+```lua
+  -- Calculate the vector from the hero to the point by multiplying
+  -- their forward vector (length 1) with the desired distance.
+  local heroToPoint = hero:GetForwardVector() * 100
+  -- Calculate world position  of the item by adding the vector
+  -- from hero to point to the world position of the hero
+  local itemPos = hero:GetAbsOrigin() + heroToPoint
```
Comment (ark120202) [inline _articles/scripting/vector-math.md:135]:
```suggestion
-- Calculate the vector from the hero to the point by multiplying
-- their forward vector (length 1) with the desired distance.
local heroToPoint = hero:GetForwardVector() * 100
-- Calculate world position  of the item by adding the vector
-- from hero to point to the world position of the hero
local itemPos = hero:GetAbsOrigin() + heroToPoint
```

**Pull request merged**